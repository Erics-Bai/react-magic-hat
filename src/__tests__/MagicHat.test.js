/* eslint-disable no-debugger */
import React from 'react'
import {mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import MagicHat from '../MagicHat'

class Dummy extends React.Component {
  render() {
    return 'hello'
  }
}

describe('MagicHat', () => {
  it('should match the snapshot', () => {
    const wrapper = mount(<MagicHat renderFrame={() => <Dummy />} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('invoke startAnimation', () => {
    const mockCallback = jest.fn()

    mount(
      <MagicHat renderFrame={() => <Dummy />} onStartAnimation={mockCallback} />
    )

    expect(mockCallback.mock.calls.length).toBe(1)
  })
})
