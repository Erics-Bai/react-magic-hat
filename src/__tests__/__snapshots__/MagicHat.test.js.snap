// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicHat should match the snapshot 1`] = `
<MagicHat
  renderFrame={[Function]}
>
  <MagicHatNoState
    activePage={1}
    flipMoveOptions={[Function]}
    moveSeed=""
    onEndAnimation={[Function]}
    onStartAnimation={[Function]}
    pages={
      Array [
        Object {
          "id": "",
          "page": 1,
          "state": Object {},
        },
      ]
    }
    renderFrame={[Function]}
  >
    <FlipMovePropConverter
      appearAnimation={
        Object {
          "from": Object {
            "transform": "scaleX(0) scaleY(1)",
            "transformOrigin": "50% 50%",
          },
          "to": Object {
            "transform": "",
          },
        }
      }
      delay={0}
      disableAllAnimations={false}
      duration={400}
      easing="cubic-bezier(0.39, 0.575, 0.565, 1)"
      enterAnimation={
        Object {
          "from": Object {
            "opacity": 0.1,
            "transform": "scaleX(0) scaleY(0.4)",
          },
          "to": Object {
            "transform": "",
          },
        }
      }
      getPosition={[Function]}
      leaveAnimation={
        Object {
          "from": Object {
            "transform": "",
            "transformOrigin": "50% 50%",
          },
          "to": Object {
            "opacity": 0.1,
            "transform": "scaleX(0) scaleY(0.8)",
          },
        }
      }
      leaveAnimationDefault={
        Object {
          "from": Object {
            "transform": "",
          },
          "to": Object {
            "opacity": 0.1,
            "transform": "scaleX(0) scaleY(0.8)",
          },
        }
      }
      maintainContainerHeight={false}
      onFinishAll={[Function]}
      onStartAll={[Function]}
      staggerDelayBy={0}
      staggerDurationBy={0}
      typeName={null}
      verticalAlignment="top"
    >
      <FlipMove
        appearAnimation={
          Object {
            "from": Object {
              "transform": "scaleX(0) scaleY(1)",
              "transformOrigin": "50% 50%",
            },
            "to": Object {
              "transform": "",
            },
          }
        }
        delay={0}
        delegated={
          Object {
            "leaveAnimationDefault": Object {
              "from": Object {
                "transform": "",
              },
              "to": Object {
                "opacity": 0.1,
                "transform": "scaleX(0) scaleY(0.8)",
              },
            },
            "style": Object {
              "position": "relative",
            },
          }
        }
        disableAllAnimations={false}
        duration={400}
        easing="cubic-bezier(0.39, 0.575, 0.565, 1)"
        enterAnimation={
          Object {
            "from": Object {
              "opacity": 0.1,
              "transform": "scaleX(0) scaleY(0.4)",
            },
            "to": Object {
              "transform": "",
            },
          }
        }
        getPosition={[Function]}
        leaveAnimation={
          Object {
            "from": Object {
              "transform": "",
              "transformOrigin": "50% 50%",
            },
            "to": Object {
              "opacity": 0.1,
              "transform": "scaleX(0) scaleY(0.8)",
            },
          }
        }
        maintainContainerHeight={false}
        onFinishAll={[Function]}
        onStartAll={[Function]}
        staggerDelayBy={0}
        staggerDurationBy={0}
        typeName={null}
        verticalAlignment="top"
      >
        <Dummy
          key=".$1"
        >
          hello
        </Dummy>
      </FlipMove>
    </FlipMovePropConverter>
  </MagicHatNoState>
</MagicHat>
`;
